src/components/semver/semver-directive.js

```diff
@@ -12,19 +12,35 @@
             if (!inputValue) {
               return inputValue;
             }
-
-            var r = /^(\d+)\.(\d+)\.(\d+)\.(\d+)$/;
-            if (!r.test(inputValue)) {
-              return inputValue;
-            }
-
-            // convert 4 part version to semver (1.2.3.4 to 1.2.3-4)
-            var transformedInput = inputValue.replace(r, '$1.$2.$3-$4');
-            if (transformedInput !== inputValue) {
-              modelCtrl.$setViewValue(transformedInput);
-              modelCtrl.$render();
+            
+            var z = /^(\d+)\.(\d+)\.?(\d+)?\.?(\d)?$/;
+            var result = '';
+            if (z.test(inputValue))
+            {
+              var z2 = /^(\d+)\.(\d+)$/;
+              if (z2.test(inputValue))
+                  result = inputValue.replace(z2, '$1.$2.0-0');
+              else 
+              {
+                var z3 = /^(\d+)\.(\d+)\.(\d+)$/;
+                if (z3.test(inputValue))
+                    result = inputValue.replace(z3, '$1.$2.$3-0');
+                else {
+                  var z4 = /^(\d+)\.(\d+)\.(\d+)\.(\d+)$/;
+                  if (z4.test(inputValue))
+                    result = inputValue.replace(z4, '$1.$2.$3-$4');    
+                }
+              }
+                
+              if (result !== '')
```

Comment:
Any specific reason for checking for not empty vs not the same as the initial input value? I'd need to look up why we were doing that. I'd assume so we didn't trigger an extra digest (update cycle)

Reply:
No reason, I copied this idea from existing code, but I don't think it even needs to be checked as the code above should never fail (otherwise the first (4x \d) guard wouldn't go through).